auth.go

```diff
@@ -440,6 +378,101 @@ func authenticate(
 	return &respd.Data, nil
 }
 
+func tryAuth(
```

Comment:
This function builds auth request data - for instance creates JWT token. We can retry this function invocation for specific error codes which will cause token recreation (with new expiration).